src/plugins/reporter-upload.c

```diff
@@ -33,32 +32,48 @@ static char *ask_url(const char *message)
     return url;
 }
 
-static int create_and_upload_archive(
-                const char *dump_dir_name,
-                map_string_t *settings)
+static int interactive_upload_file(const char *url, const char *file_name, char **remote_name)
 {
-    int result = 1; /* error */
+    post_state_t *state = new_post_state(POST_WANT_ERROR_MSG);
+    state->username = getenv("Upload_Username");
+    char *password_inp = NULL;
+    if (state->username != NULL && state->username[0] != '\0')
+    {
+        /* Load Password only if Username is configured, it doesn't make */
+        /* much sense to load Password without Username. */
+        state->password = getenv("Upload_Password");
```

Comment:
What if user configures scp to use keys? There might be username set but no password is needed. Also using CURL to upload stuff via SCP is not good at all, would be much better to just use SCP so user can configure it as usual and not fiddle with CURL/abrt specific configuration.